avocado/core/nrunner/app.py

```diff
@@ -114,6 +115,10 @@ def __init__(self, echo=print, prog=None, description=None):
         self._class_commands_method = self._get_commands_method()
         self._setup_parser(prog, description)
 
+    def echo(self, message):
+        message['time'] = time.monotonic()
+        self._echo(message)
+
```

Comment:
Usually it is "strange" to have a method with the same name as an argument. This leads to confusing when using the API:

```python
r = Runner(echo=print)
r.echo # this will be completely different from what passed by the user.
```

What about renaming `s/echo/send_message/`? Or rename the `echo` argument to `echo_method`.

Also, if an argument is being used it is ok to leave as public, since it was passed during the initialization.